### ISSUE TITLE
FieldCanBeLocal static field incorrectly patched

### ISSUE THREAD

**Issue by delanym** [google/error-prone#3814]
Where
`private static String logFileNameBase;`
is only used locally, the patch replaces 
`Logger.logFileNameBase = Logger.logFolderName + name;`
with
`String Logger.logFileNameBase = Logger.logFolderName + name;`
when it should be
`String logFileNameBase = Logger.logFolderName + name;`

**Comment by rickie** [comment]
I'm wondering if this is actually an _incorrect_ patch. Shouldn't the `UnnecessarilyFullyQualified` [check](https://errorprone.info/bugpattern/UnnecessarilyFullyQualified) remove the `Logger.` prefix here if it is not required to have?

From a quick skim of the implementation it seems like only `this.` is [being stripped](https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/FieldCanBeLocal.java#L279).

**Comment by delanym** [comment]
@rickie no. Its not being "fully" qualified. Some devs want their static constants/methods qualified rather than imported directly with a static import.
